core/amber/src/main/python/core/runnables/data_processor.py

```diff
@@ -29,9 +29,38 @@ def run(self) -> None:
         self._running.set()
         self._switch_context()
         while self._running.is_set():
+            self.process_state()
```

Comment:
It is either a state or a tuple to process. please change it to:
```
if xxx:
     self.process_state()
else:
     self.process_tuple()
```

Reply:
Fixed.